### ISSUE TITLE
Parser fails on nested output filters

### ISSUE THREAD

**Issue by goldsky** [modxcms/revolution#14095]
### Summary
Related issue: https://github.com/modxcms/revolution/issues/13250 .
Parser fails to process nested output filters, and returns truncated tags.

### Step to reproduce
Create a chunk called 'rowChunk' with this content:
```
[[+id:is=`1`:then=`[[+id]]`:else=`
    [[+id:is=`2`:then=`[[+id]]`:else=`
        [[+id:is=`3`:then=`[[+id]]`:else=`More`]]
    `]]
`]]
```

Then use the chunk as this:
```
[[!$rowChunk?
&id=`2`
]]
```
Try with ``` &id=`3` ```
### Observed behavior
It returns
```
More`]] `]] 
```

### Expected behavior
It should return the defined value.

### Environment
MODX 2.6.5-pl

**Comment by Jako** [comment]
@bezumkin Do you have an idea, what could happen here?

**Comment by bezumkin** [comment]
@Jako Problem is [here](https://github.com/modxcms/revolution/blob/e59f4e0de39adf5b3091ff0281d3062f4cdadb2f/core/model/modx/filters/modinputfilter.class.php#L54-L57)

This expression [couldn't handle nested conditions](https://regex101.com/r/vzttSe/1). Here what is happening:
![screenshot 2019-01-17 at 17 27 39](https://user-images.githubusercontent.com/1257284/51312188-4a391580-1a7d-11e9-8b67-390a74d0b11b.png)

I can't fix it and, frankly, I don't use anything like this because of Fenom.

**Comment by JoshuaLuckers** [comment]
I will give it a try in the upcoming days 👍

**Comment by goldsky** [comment]
I am not sure if this is related, but how about changing the MODX's parsing order for nested tags, from inner outwards to outer inwards?

That means, the logic condition should start from the first tag and stop immediately once it meets the condition, without going further into the deeper nested tag(s).

**Comment by bezumkin** [comment]
@goldsky MODX parser always process tags inside out. Changing this will brake everything.

**Comment by goldsky** [comment]
Yeah, I know that irony.
It shouldn't be at the first place.

**Comment by JoshuaLuckers** [comment]
I submitted a PR to fix this issue. Feel free to test the changes and provide feedback.

**Comment by Ibochkarev** [comment]
Fixed #14458 
@goldsky Thanks
@modxbot close
